libcloudforensics/providers/aws/internal/s3.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Bucket functionality."""
+
+from typing import TYPE_CHECKING, Dict, Optional, Any
+
+from libcloudforensics import errors
+from libcloudforensics.providers.aws.internal import common
+
+if TYPE_CHECKING:
+  # TYPE_CHECKING is always False at runtime, therefore it is safe to ignore
+  # the following cyclic import, as it it only used for type hints
+  from libcloudforensics.providers.aws.internal import account  # pylint: disable=cyclic-import
+
+
+class S3:
+  """Class that represents AWS S3 storage services.
+
+  Attributes:
+    aws_account (AWSAccount): The account for the resource.
+    name (str): The name of the bucket.
+    region (str): The region in which the bucket resides.
+  """
+
+  def __init__(self,
+               aws_account: 'account.AWSAccount') -> None:
+    """Initialize the AWS S3 resource.
+
+    Args:
+      aws_account (AWSAccount): The account for the resource.
+
+    """
+
+    self.aws_account = aws_account
+
+  def CreateBucket(
+      self,
+      name: str,
+      region: Optional[str] = None,
+      acl: Optional[str] = 'private') -> Dict[str, Any]:
```

Comment:
The `Optional` type hint can be removed here since it's assigned a default value.
In general, for named parameters that default to `None`, we use the `Optional` type hint. All other cases, we don't.